lib/scr/bloc/dice_bloc.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'i_dice_bloc.dart';
+
+class DiceBloc extends IDiceBloc {
+  final int _numberOfDice = 4;
+  StreamController<List<int>> _streamController = StreamController<List<int>>();
+  StreamController<int> _scoreValue = StreamController<int>();
+
+  @override
+  void dispose() {
+    _streamController.close();
+    _scoreValue.close();
+  }
+
+  void roll() {
+    List<int> _listAux = [];
+    var sum = 0;
+    for (var i = 0; i < numberOfDice; i++) {
+      _listAux.add(Random().nextInt(6) + 1);
+      sum += _listAux[i];
+    }
+    _streamController.sink.add(_listAux);
+    _scoreValue.sink.add(sum);
+  }
+
+  List<int> defaultValue() {
+    var sum = 0;
+    List<int> _listAux = [];
+    for (var i = 0; i < numberOfDice; i++) {
+      _listAux.add(Random().nextInt(6) + 1);
+      sum += _listAux[i];
+    }
+    _scoreValue.sink.add(sum);
+    return _listAux;
+  }
```

Comment:
Here you are doing similar code in both functions (roll and defaultValue),try to separate the portion of code that repeats into a separate function and then call it in each method where it is used.

Reply:
Fixed